SingularityService/src/main/java/com/hubspot/singularity/auth/SingularityAuthorizationHelper.java

```diff
@@ -138,13 +143,17 @@ public void checkForAuthorization(SingularityRequest request, Optional<Singulari
     checkUnauthorized(user.isPresent(), "user must be present");
 
     final Set<String> userGroups = user.get().getGroups();
+    final Set<String> readWriteGroups = request.getReadWriteGroups().or(Collections.<String>emptySet());
+    if (request.getGroup().isPresent()) {
+      readWriteGroups.add(request.getGroup().get());
```

Comment:
`Collections.emptySet()` is immutable, unfortunately, and we generally shouldn't be mutating stuff in `SingularityRequest`. I'd suggest just using `Sets.union()` to generate a the union here.